src/v2/api/services/PermissionService.spec.ts

```diff
@@ -0,0 +1,349 @@
+import { PermissionService } from './PermissionService';
+import { RoleRepository } from '../../database/repositories/RoleRepository';
+import { Test } from '@nestjs/testing';
+
+describe('PermissionService', () => {
+  let permissionService: PermissionService;
+
+  beforeEach(async () => {
+    const testingModule = await Test.createTestingModule({
+      providers: [PermissionService],
+    }).useMocker((token) => {
+      if (token === RoleRepository) {
+        return {};
+      }
+    })
+      .compile();
+
+    permissionService = testingModule.get(PermissionService);
+  });
+
+  describe('hasGrantPermission', () => {
+    it('should false if different roots', () => {
+      const grant = 'one';
+      const permission = 'two';
+
+      const result = permissionService.hasGrantPermission(permission, grant);
+
+      expect(result).toBe(false);
+    });
+
+    it('should false if same roots, but different second child', () => {
```

Comment:
I'm not sure if the logic of this method requires so many different cases. Check the cyclomatic complexity of this method, if it matches the number of your tests, it's OK.

Reply:
Thanks for your review! I checked the cyclomatic complexity (6) and removed unnecessary tests